main.go

```diff
@@ -2,110 +2,148 @@ package main
 
 import (
 	"context"
-	"net/http"
+	"flag"
+	"fmt"
 	"os"
 
+	producer "github.com/asecurityteam/component-producer"
 	"github.com/asecurityteam/ipam-facade/pkg/assetfetcher"
 	"github.com/asecurityteam/ipam-facade/pkg/assetstorer"
 	"github.com/asecurityteam/ipam-facade/pkg/domain"
 	v1 "github.com/asecurityteam/ipam-facade/pkg/handlers/v1"
 	"github.com/asecurityteam/ipam-facade/pkg/ipamfetcher"
+	"github.com/asecurityteam/ipam-facade/pkg/randomnumbergenerator"
 	"github.com/asecurityteam/ipam-facade/pkg/sqldb"
 	"github.com/asecurityteam/serverfull"
 	"github.com/asecurityteam/settings"
 )
 
-func main() {
-	ctx := context.Background()
-	source, err := settings.NewEnvSource(os.Environ())
+type producerConfig struct {
```

Comment:
Question: why are we creating this `producerConfig` struct? It looks like `producer.Config` already has a `Name()` func which returns "producer".